### PR TITLE
UTH-21: Get application profile

### PR DESCRIPTION
**Pull request by momentiris**
- Adds endpoint for fetching application profile by contact code

### CODE REVIEW

**Review by mittistormen — APPROVED** [review]

File: src/services/lease-service/tests/routes/contacts.test.ts [hunk 1]
```diff
@@ -139,3 +140,46 @@ describe('GET /contacts/search', () => {
...
+
+    const res = await request(app.callback()).get(
+      '/contacts/1234/application-profile'
+    )
+
+    expect(res.status).toBe(200)
+    expect(() =>
+      leasing.GetApplicationProfileResponseDataSchema.parse(res.body.content)
+    ).not.toThrow()
+    expect(res.body.content).toEqual(expect.objectContaining({ id: 1 }))
```
Comment (momentiris) [inline src/services/lease-service/tests/routes/contacts.test.ts:183]:
```suggestion
```
Redundant

**Review by driatic — APPROVED** [review]
Sleek! Looks great 🥇

File: src/services/lease-service/routes/contacts.ts [hunk 2]
```diff
@@ -433,4 +435,66 @@ export const routes = (router: KoaRouter) => {
...
+      }
+
+      ctx.status = 500
+      ctx.body = { error: 'unknown', ...metadata }
+      return
+    }
+
+    ctx.status = 200
+    ctx.body = {
+      content: profile.data satisfies GetApplicationProfileResponseData,
```
[the remark this answers is not in the document]
Comment (momentiris) [reply src/services/lease-service/routes/contacts.ts:496]: `satisfies` is just a TS keyword that instructs TS to "make sure the left hand part complies with the right hand part". But without casting it, as you would do using `as`. So

File: src/services/lease-service/routes/contacts.ts [hunk 3]
```diff
@@ -433,4 +435,66 @@ export const routes = (router: KoaRouter) => {
...
+   *     responses:
+   *       200:
+   *         description: Successfully retrieved application profile.
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: object
+   *               properties:
+   *                 data:
+   *                   type: object
```
[the remark this answers is not in the document]
Comment (momentiris) [reply src/services/lease-service/routes/contacts.ts:463]: agreed!

**Pull request merged**